### PR TITLE
Allure Testops: add fields

### PR DESCRIPTION
**Pull request by andrey-brovko**

### CODE REVIEW

File: GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs [hunk 1]
```diff
@@ -65,20 +69,126 @@ public CreateTestCaseRequestExtended BuildCaseRequest(Scenario scenario, IFeatur
...
+    private List<CustomFieldItem> AddCustomFields(Scenario scenario, IFeatureFile featureFile)
+    {
+        var result = new List<CustomFieldItem>();
+        AddFeatureCustomFiled(featureFile, result);
+        AddEpicCustomFiled(featureFile, result);
+        AddComponentCustomFiled(featureFile, scenario, result);
+        return result;
+    }
+
+    private void AddComponentCustomFiled(IFeatureFile featureFile, Scenario scenario, List<CustomFieldItem> result)
```
Comment (afomina-quantori) [inline GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs:90]: looks like spellling error,  should be renamed to AddComponentCustomField

File: GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs [hunk 2]
```diff
@@ -65,20 +69,126 @@ public CreateTestCaseRequestExtended BuildCaseRequest(Scenario scenario, IFeatur
...
+    {
+        var settingsComponent = _allureTestOpsSettings.Component;
```
Comment (afomina-quantori) [inline GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs:92]: let's move the variable declaration closer to the place where it is used

File: GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs [hunk 3]
```diff
@@ -65,20 +69,126 @@ public CreateTestCaseRequestExtended BuildCaseRequest(Scenario scenario, IFeatur
...
+
+        if (!_customFilesValues.ContainsKey(componentField.CustomField.Id))
+        {
+            _customFilesValues.Add(componentField.CustomField.Id, values.ToList());
+        }
+
+        return componentField;
+    }
+
+    private void AddCustomFiled(List<CustomFieldItem> result, CustomFieldSchemaContent customField, string cfValue)
```
Comment (afomina-quantori) [inline GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs:138]: seems like spelling error

File: GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs [hunk 4]
```diff
@@ -65,20 +69,126 @@ public CreateTestCaseRequestExtended BuildCaseRequest(Scenario scenario, IFeatur
...
+    {
+        var value = _customFilesValues[customField.CustomField.Id].FirstOrDefault(item => item.Name.Equals(cfValue));
```
Comment (afomina-quantori) [inline GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs:140]: maybe, the _customFilesValues should be renamed into _customFieldValues?

File: Quantori.AllureTestOpsClient/Model/CustomFieldItem.cs [hunk 5]
```diff
@@ -2,7 +2,7 @@
 {
     public class CustomFieldItem
```
Comment (afomina-quantori) [inline Quantori.AllureTestOpsClient/Model/CustomFieldItem.cs:3]: Could you please add a xml summary to this class?

File: GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs (hunk 2 above)
Comment (andrey-brovko) [reply GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs]: Fixed

File: GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs (hunk 3 above)
Comment (andrey-brovko) [reply GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs:137]: Fixed

File: Quantori.AllureTestOpsClient/Model/CustomFieldItem.cs (hunk 5 above)
Comment (andrey-brovko) [reply Quantori.AllureTestOpsClient/Model/CustomFieldItem.cs:3]: This is just a DTO

File: GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs (hunk 4 above)
Comment (andrey-brovko) [reply GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs]: Done

**Review by afomina-quantori — APPROVED** [review]

**Pull request merged**